definitions/features/installer.rb

```diff
@@ -105,13 +105,19 @@ def installer_arguments
     installer_args = ''
 
     if feature(:foreman_proxy) &&
-       feature(:foreman_proxy).with_content?
+       feature(:foreman_proxy).with_content? &&
+       !feature(:katello_install)
       installer_args += ' --disable-system-checks'
     end
 
     unless check_min_version('foreman', '2.1') || check_min_version('foreman-proxy', '2.1')
       installer_args += ' --upgrade' if can_upgrade?
     end
+
+    if feature(:instance).upstream_install
+      installer_args += ' --tuning development'
```

Comment:
I was able to install the Katello 4.4 with default tuning profile I think the default tuning requirement of 20GB changed in 4.5 or later,
~~~
2022-08-05 10:32:40 [NOTICE] [root] Running installer with log based terminal output at level NOTICE.
2022-08-05 10:32:40 [NOTICE] [root] Use -l to set the terminal output log level to ERROR, WARN, NOTICE, INFO, or DEBUG. See --full-help for definitions.
Insufficient memory for tuning size
Tuning profile 'default' requires at least 20 GB of memory and 4 CPU cores
                                      [FAIL]
~~~

How we should handle the situation where existing profile was default but we changed the defaults and now system fails to upgrade?

Reply:
That is a good question, we are faced with two choices that I see:

 1) Ensure the upgrade happens no matter what the case
 2) Throw an error if the system is out of compliance with the requirements so that the user can fix that before upgrading

@evgeni @ekohl Would appreciate your thoughts here

Reply:
I vote 2?

Reply:
A question is, if we want to ensure the upgrade continues should we run this with `--dont-save-answers` to only set the profile during runtime and not change it for the user permanently.

Reply:
IMHO this is one of those things we can't answer so I'd also vote option 2. In the future if we do need to raise requirements the user should also be aware of this. They may be fine with increasing the server size rather than suffering reduced performance.

The best way to detect this would be to use a specific exit code. Then foreman-maintain can check for it. The good news is that we already do: codes 102 (insufficient CPU cores) and 103 (insufficient memory) are used.

Reply:
I have added the Check which runs the installer in noop mode and throws error if any in pre upgrade stage. As I am at it and as per the IRC discussion changed the unless condition to depend on `foreman-installer` package instead of `foreman` and `foreman-proxy` to include `--upgrade` flag.